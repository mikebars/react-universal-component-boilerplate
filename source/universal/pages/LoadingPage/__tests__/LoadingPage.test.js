/* @flow */

import React from 'react'
import { HelmetProvider } from 'react-helmet-async'
import { Router } from 'react-router-dom'
import createMemoryHistory from 'history/createMemoryHistory'
import renderer from 'react-test-renderer'
import { mount, shallow, render } from 'enzyme'

import { LoadingPage } from 'universal/pages/LoadingPage/LoadingPage'

test('LoadingPage - snapshot test', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <HelmetProvider>
      <Router history={history}>
        <LoadingPage {...props} />
      </Router>
    </HelmetProvider>
  )

  const tree = renderer.create(element).toJSON()

  expect(tree).toMatchSnapshot()
})

test('LoadingPage - enzyme shallow render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <HelmetProvider>
      <Router history={history}>
        <LoadingPage {...props} />
      </Router>
    </HelmetProvider>
  )

  const wrapper = shallow(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('LoadingPage - enzyme full render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <HelmetProvider>
      <Router history={history}>
        <LoadingPage {...props} />
      </Router>
    </HelmetProvider>
  )

  const wrapper = mount(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('LoadingPage - enzyme static render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <HelmetProvider>
      <Router history={history}>
        <LoadingPage {...props} />
      </Router>
    </HelmetProvider>
  )

  const wrapper = render(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})
