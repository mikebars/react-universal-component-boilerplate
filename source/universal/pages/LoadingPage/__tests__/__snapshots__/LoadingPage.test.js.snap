// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage - snapshot test 1`] = `
<section
  className="Page-o0cu63-0 cQYYlG"
>
  <section
    className="Center-s1yx1lyo-0 lgKgCZ"
  >
    <span
      className="Text-ia49n9-0 cKyjWr"
      fontSize="10vw"
    >
      Loading...
    </span>
  </section>
</section>
`;
